src/py/mat3ra/made/basis.py

```diff
@@ -76,7 +76,17 @@ def to_crystal(self):
         self.coordinates.map_array_in_place(self.cell.convert_point_to_crystal)
         self.units = AtomicCoordinateUnits.crystal
 
-    def add_atom(self, element="Si", coordinate=[0.5, 0.5, 0.5]):
+    def add_atom(self, element="Si", coordinate=None, force=False):
```

Comment:
The function signature here should be aware of whether the units of the coordinate are cartesian or crystal. We can add `use_cartesian` or something like that.

Before lines 90 and 91 we need to check and confirm that the current units are compatible - i.e. if units are cartesian, we should add cartesian_coordinate, otherwise - crystal